### PR TITLE
fetch data for PDP

### PR DESCRIPTION
**Pull request by jshimko**
This PR relies on https://github.com/reactioncommerce/reaction/pull/4218 to work (which should hopefully be merged soon)

Updates:
- fetch product item data using URL slug or ID
- update any PDP components with mapped data
- update mock data to match GraphQL data

### CODE REVIEW

**Comment by spencern** [comment]
@jshimko How close are we to having a PR with this?

**Comment by jshimko** [comment]
This is still depending on the catalog schema PR in the main Reaction repo (https://github.com/reactioncommerce/reaction/pull/4218), but it's ready to be tested/reviewed.  

```
reaction init -b refactor-4196-aldeed-catalog-schema-changes catalog-schema
cd catalog-schema
docker-compose up -d --build
```

Then start up this PR branch and everything should work.  You should see the PDP is now completely populated by the GraphQL API data.

File: src/lib/apollo/initApolloServer.js [hunk 1]
```diff
@@ -11,17 +12,34 @@ if (!process.browser) {
   global.fetch = fetch;
 }
 
+// error logging for Apollo Link
+const errorLink = onError(({ graphQLErrors, networkError }) => {
+  if (graphQLErrors) {
+    graphQLErrors.forEach(({ message, locations, path }) => {
+      // eslint-disable-next-line no-console
+      console.log(`[GraphQL error]: Message: ${message}, Location: ${JSON.stringify(locations)}, Path: ${JSON.stringify(path)}`);
```
Comment (nnnnat) [inline src/lib/apollo/initApolloServer.js]: could we use the logger util for something like this?
https://github.com/reactioncommerce/reaction-next-starterkit/blob/master/src/lib/logger/index.js

Comment (jshimko) [reply src/lib/apollo/initApolloServer.js]: Well, maybe... but that logger is using `chalk` (for terminal colors) and these particular errors can happen on the server _and_ the client.  I think we probably need a more isomorphic solution for this use case.  We could certainly use [Bunyan](https://github.com/trentm/node-bunyan) for that.  Then we'd have all of the same API's we're used to in Reaction (for piping logs to other places, etc).

**Comment by jshimko** [comment]
@nnnnat FYI, it's hidden in the Github UI now because I pushed changes, but I did respond to the logging questions.

File: src/lib/apollo/initApolloServer.js (hunk 1 above)
Comment (nnnnat) [reply src/lib/apollo/initApolloServer.js]: yeah, I wasn't sure if our `lib/logger` would work in these cases. I'm good with just doing the `console.error` here for now and maybe we revisit our logger soon in another PR

**Comment by nnnnat** [comment]
@spencern  @aldeed  I think this is good 2 go, but should we wait on [PR 4218](https://github.com/reactioncommerce/reaction/pull/4218) before we merge this one?

**Pull request merged**